receiver/redisreceiver/receiver.go

```diff
@@ -0,0 +1,71 @@
+package redisreceiver
+
+import (
+	"sync"
+	"time"
+
+	"github.com/go-redis/redis/v7"
+	"github.com/open-telemetry/opentelemetry-collector/component"
+	"github.com/open-telemetry/opentelemetry-collector/consumer"
+	"github.com/open-telemetry/opentelemetry-collector/oterr"
+	"github.com/open-telemetry/opentelemetry-collector/receiver"
+	"go.uber.org/zap"
+)
+
+type redisReceiver struct {
+	sync.Mutex
+	logger         *zap.Logger
+	config         *Config
+	consumer       consumer.MetricsConsumer
+	intervalRunner *intervalRunner
+
+	startOnce sync.Once
+	stopOnce  sync.Once
+}
+
+var _ receiver.MetricsReceiver = (*redisReceiver)(nil)
+
+func newRedisReceiver(logger *zap.Logger, config *Config, consumer consumer.MetricsConsumer) *redisReceiver {
+	return &redisReceiver{
+		logger:   logger,
+		config:   config,
+		consumer: consumer,
+	}
+}
+
+func (r redisReceiver) Start(host component.Host) error {
+	r.Lock()
```

Comment:
I modeled this on the other receivers but would like to learn more about why the mutex and startOnce are needed.

Reply:
It should not be needed. `Start` is called once only.

@pjanotti any thoughts on this? I saw the same pattern in `signalfxreceiver`, so perhaps there is something I am missing.

(We do have "once" protection in opencensusreceiver, but there it is needed because it implements trace and metrics receiver via single instance and internal `start` function may be called twice. Everywhere else I do not think we need such protection).

Reply:
Ok, will remove for now. Thanks.